packages/automerge-repo/src/DocHandle.ts

```diff
@@ -448,14 +477,58 @@ export class DocHandle<T> //
   }
 }
 
+function optionsWithGlobalMetadata<T>(
+  options: DocHandleChangeOptions<T>,
+  globalMetadata: ChangeMetadata
+): A.ChangeOptions<T> {
+  let metadata = { ...globalMetadata }
+
+  if (options.metadata) {
+    if (!metadata) {
+      metadata = {}
+    }
+
+    Object.assign(metadata, options.metadata)
```

Comment:
Per https://github.com/automerge/automerge-repo/pull/267/files#r1449148679 I think we should check that the values here are only of the allowed types and throw if not.